### PR TITLE
Check if IP is in subnet before forcing IP on OpenStack port

### PR DESCRIPTION
**Pull request by ghost**
Users have complained that the migration cannot happen if the IP address of the migrated VM is not in the destination subnet in OpenStack. This pull request checks that the IP address is in the subnet range of the network, before adding the fixed IP address to the port. This will allow creating the VM and let the consequence befall on user.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: wrapper/hosts.py [hunk 1]
```diff
@@ -418,9 +419,21 @@ def handle_finish(self, data, state):
...
             if 'ip_address' in nic:
-                port_cmd.extend([
-                    '--fixed-ip', 'ip-address=%s' % nic['ip_address'],
-                    ])
+                subnets_cmd = [
+                    'subnet', 'list',
+                    '--network', nic['destination'],
+                    '-f', 'json'
+                ]
+                subnets = json.load(self._run_openstack(subnets_cmd, data))
```
Comment (nyoxi) [inline wrapper/hosts.py]: check for `None` before calling `loads`

File: wrapper/hosts.py [hunk 2]
```diff
@@ -525,6 +539,25 @@ def validate_data(self, data):
...
         return data
 
+    def _check_ip_in_network(self, ipaddr, network):
+        [netaddr, netsize] = network.split('/')
+        netsize = int(netsize)
+        ip_prefix_bin = self._get_prefix_bin(ipaddr, netsize)
+        net_prefix_bin = self._get_prefix_bin(netaddr, netsize)
+        if ip_prefix_bin == net_prefix_bin:
+            return True
+
```
Comment (nyoxi) [inline wrapper/hosts.py]: remove this empty line

File: wrapper/hosts.py [hunk 3]
```diff
@@ -525,6 +539,25 @@ def validate_data(self, data):
                 hard_error('Missing mac address in one of network mappings')
         return data
 
+    def _check_ip_in_network(self, ipaddr, network):
```
Comment (nyoxi) [inline wrapper/hosts.py:542]: it would be nice to have at least a few tests for this and the following two functions

**Review by nyoxi — COMMENTED** [review]
Once done, could you please squash the commits together? There's really no need to keep this small change separated into so many commits.

**Comment by nyoxi** [comment]
ci test please

**Pull request merged**